statsd/options.go

```diff
@@ -419,7 +417,10 @@ func WithContainerID(id string) Option {
 // WithCardinality sets the tag cardinality of the metric.
 func WithCardinality(card Cardinality) Option {
 	return func(o *Options) error {
-		o.tagCardinality = card
+		if !card.isValid() {
```

Comment:
Here we ensure user can never pass invalid cardinality to the client during construction.